resources/core/charts/apiserver-proxy/files/generate-certs.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+set -e
+# if running on Gardener create Certificate CR
+# else proceed 'old' way
+if [ "$DOMAIN" = "" ]; then
+{{ if .Values.global.domainName }}
+{{ if .Values.global.environment.gardener }}
+cat <<EOF | kubectl apply -f -
+apiVersion: cert.gardener.cloud/v1alpha1
+kind: Certificate
+metadata:
+  name: apiserver-proxy-tls-cert
+  namespace: kyma-system
+spec:
+  commonName: "apiserver.{{ trimPrefix "*." .Values.global.domainName }}"
+  secretName: "{{ template "name" . }}-tls-cert"
+EOF
+
+  SECONDS=0
+  END_TIME=$((SECONDS+600)) #600 seconds = 10 minutes
+  while [ ${SECONDS} -lt ${END_TIME} ];do
+    STATUS="$(kubectl get -n {{ .Release.Namespace }} certificate.cert.gardener.cloud {{ template "name" . }}-tls-cert -o jsonpath='{.status.state}')"
+    if [ "${STATUS}" = "Ready" ]; then
+      break
+    fi
+    echo "Waiting for Certicate generation, status is ${STATUS}"
+    sleep 10
+  done
+
```

Comment:
We need to check properly if it's ready. After 10 mins it can be still in error state, then it try process but should exit instead